### ISSUE TITLE
Do not produce dependency items for the quick pulse calls

### ISSUE THREAD

**Issue by SergeyKanzhelev** [Microsoft/ApplicationInsights-dotnet-server#196]
>Using app insights in VS I keep seeing these “Dependecy /QuickPulseService.svc” lines show up
 
>![image](https://cloud.githubusercontent.com/assets/9950081/19975863/ca8b97ec-a1aa-11e6-9bb6-9e0970c34680.png)

>What is this? It’s kind of annoying that it shows up every 5 seconds!

VS shows all telemetry items – filtered and not filtered. One of the reasons for this is that we wanted to show “sampled out”  requests in the local search. If QP generate dependencies events – SDK spends many cycles to allocate and initialize those. So this is clearly a wasted effort.

I think the solution is to filter those calls in Dependency Tracking module, not in the filter.

**Comment by SergeyKanzhelev** [comment]
I think both should be happening:

1. Filtering in Dependency Collector by default
2. As @gardnerjr pointed - this should be included into filtering processor:

``` csharp
if (IsQuickPulse(item))
{
    TelemetryDebugWriter.WriteTelemetry(item, "QuickPulse Filtering");
    return;
}
```
It will ends up with the prefix: `Application Insights Telemetry (filtered by QuickPulse Filtering): ` in the output that will be marked by VSIX

**Comment by gardnerjr** [comment]
the above was just what i did locally in a filter to make them go away in one of my projects and test the filtering, but yes, there should be some filtering by the AI SDK itself, it shouldn't be considering outbound quickpulse calls to be "dependencies" in *my* application.  

from the SDK side, i don't think there should even be the `TelemetryDebugWriter` call, as it isn't real telemetry.  it should get filtered out silently, it isn't telemetry being sent by my application and not clog up output windows or debug tools.

**Comment by cijothomas** [comment]
The quickpulse dependency is showing in Ibiza portal as well..

**Comment by gardnerjr** [comment]
@cijothomas yes, because they're getting sent, that's the bug :)  once they're sent, they show up *everywhere*.  VS search, AA, Ibiza, Draft, powerbi, etc.

**Comment by cijothomas** [comment]
yea will fix it.. too much noise this is causing while debugging.

**Comment by AlexBulankou** [comment]
We misunderstood the issue :( We thought it was only showing in VS. Yes, need to fix for 2.2 Stable.
